go/libkb/chain_link.go

```diff
@@ -962,13 +968,10 @@ func (c *ChainLink) GetSigchainV2Type() (SigchainV2Type, error) {
 func (c *ChainLink) checkServerSignatureMetadata(ckf ComputedKeyFamily) (ret keybase1.KID, err error) {
 	var serverKID, linkKID, verifyKID keybase1.KID
 
-	if jw := c.packed.AtKey("kid"); !jw.IsNil() {
-		serverKID, err = GetKID(jw)
-		if err != nil {
-			return ret, err
-		}
-	}
-
+	// PC: i'm not sure what exactly this was trying to do since
```

Comment:
How do you know this is the case?

Reply:
See here:

https://github.com/keybase/client/pull/10405/files#diff-9c47c50340e7d8d0d6f1ea9b9b1be878L597

Unpack is called right after NewChainLink, takes kid out of `packed`, puts it in `c.unpacked` after several checks.

Then in this function it was taking the kid out of `c.packed` again and comparing it to what was in `c.unpacked` again.